internal/watch/discover_test.go

```diff
@@ -14,162 +14,31 @@ func TestDiscoverModules(t *testing.T) {
 	ctx := log.ContextWithNewDefaultLogger(context.Background())
 	modules, err := DiscoverModules(ctx, []string{"testdata"})
 	assert.NoError(t, err)
-	expected := []moduleconfig.ModuleConfig{
+	expected := []moduleconfig.UnvalidatedModuleConfig{
```

Comment:
Previously this test was testing discovery and defaulting.
Now it just tests discovery. Generic defaulting tests can now be found in `moduleconfig` tests, while language specific default testing can be found in `languageplugin`